tensorflow/python/estimator/estimator.py

```diff
@@ -961,10 +961,14 @@ def get_hooks_from_the_first_device(per_device_hooks):
           assert hooks_list
           return hooks_list[0]
 
-        training_hooks = get_hooks_from_the_first_device(
-            grouped_estimator_spec.training_hooks)
-        training_chief_hooks = get_hooks_from_the_first_device(
-            grouped_estimator_spec.training_chief_hooks)
+        training_hooks = [
+            get_hooks_from_the_first_device(per_device_hook)
```

Comment:
Could you move this common logic into the `get_hooks_from_the_first_device` method since it's common for both the usages?

Reply:
OK. I have moved it into `get_hooks_from_the_first_device`. Please review it again.

And Is it ok to put my test case in the class to make `test_complete_flow_with_mode` and my case run sequentially, or should I create a separated class or file? Sorry but I am not so familiar with the test codes.